ops/framework.py

```diff
@@ -475,6 +475,23 @@ def define_event(cls, event_kind: str, event_type: 'Type[EventBase]'):
         event_descriptor.__set_name__(cls, event_kind)
         setattr(cls, event_kind, event_descriptor)
 
+    @classmethod
+    def _undefine_event(cls, event_kind: str):
+        """Remove the definition of an event on this type at runtime.
+
+        This undoes the effect of :meth:`define_event`. This is not intended
+        for use by charm authors, but rather for use by the ops library itself.
+        """
+        event_descriptor = getattr(cls, event_kind)
+        if hasattr(event_descriptor, 'framework') and event_descriptor.framework is not None:
```

Comment:
Would this be more elegant / easier to read?

```suggestion
        if (framework := getattr(event_descriptor, 'framework', None)) is not None:
```

Reply:
I really don't like the walrus operator, and feel like it almost never makes code more readable. However, I feel that's a me thing, so I'll make this change, thanks!